### PR TITLE
fix issue #485 : MessagePack queries: Exception during parsing InfluxDB version [macOS]

### PR DESCRIPTION
**Pull request by lxhoan**

### CODE REVIEW

**Comment by bednar** [comment]
Maybe it will be better use some implementation for the [Semantic Versioning](https://semver.org) - [semver4j](https://github.com/vdurmont/semver4j).

The method `checkMessagePackSupport` will be pretty simple:

```java
private boolean checkMessagePackSupport() {
    Semver version = new Semver(version(), Semver.SemverType.NPM);
    return version.isGreaterThan("1.4");
}
```

**Comment by lxhoan** [comment]
@bednar 
I don't think main project maintainers will approve for new dependencies (semver4j) if it not indispensable

**Comment by lxhoan** [comment]
there unit test added for the new version checking, it should cover MacOS case and even unrealistic cases

**Comment by lxhoan** [comment]
@fmachado , please have a look on it, tks

**Comment by fmachado** [comment]
@lxhoan I know this PR is already merged (and apologies for the delay) but as you requested, I'll add my comments.

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 1]
```diff
@@ -666,13 +669,26 @@ public boolean databaseExists(final String name) {
     public String error;
   }
 
+  private boolean checkMessagePackSupport() {
+    Matcher matcher = Pattern.compile("(\\d+\\.*)+").matcher(version());
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBImpl.java]: AFAIK, `Pattern.compile` creates an immutable object so you can reuse it. There is another regular expression that would help you so you don't have to split: `(?:(\d+)\.)(\d+)`. (you can see it working [here](https://www.regexplanet.com/share/index.html?share=yyyyy1bypyr)). You still have to convert it to int, though.

**Pull request merged**